### ISSUE TITLE
Gatsby not rendering correct CSS classes in build for the initial page request

### ISSUE THREAD

**Issue by AminEl-Bekry** [gatsbyjs/gatsby#35894]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

On our website (https://qiwio.io/), on initial page request, one of the links in the header has a missing CSS class. Once you go to a new page, the CSS class name is then rendered. This happens everytime when you go to our website. We do not experience this issue in local host, only once built and deployed. 

We used styled components, maybe the issue stems from the gatsby plugin? 

### Reproduction Link

https://github.com/AminEl-Bekry/qiwio-website

### Steps to Reproduce

1. Visit https://qiwio.io/ and look at the links in the header.
...

### Expected Result

Links should be displayed as white text and no text decoration as you will see when navigating pages. 

### Actual Result

See above.

### Environment

```shell
System:
    OS: macOS 12.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.4 - /usr/local/bin/node
    Yarn: 1.22.15 - ~/.yarn/bin/yarn
    npm: 8.10.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 102.0.5005.61
    Firefox: 101.0
    Safari: 15.4
  npmPackages:
    gatsby: ^4.15.0 => 4.15.1
    gatsby-plugin-gatsby-cloud: ^4.15.0 => 4.15.0
    gatsby-plugin-google-gtag: ^4.15.0 => 4.15.0
    gatsby-plugin-image: ^2.15.0 => 2.15.0
    gatsby-plugin-manifest: ^4.15.0 => 4.15.0
    gatsby-plugin-offline: ^5.15.0 => 5.15.0
    gatsby-plugin-react-helmet: ^5.15.0 => 5.15.0
    gatsby-plugin-sass: ^5.15.0 => 5.15.0
    gatsby-plugin-sharp: ^4.15.0 => 4.15.0
    gatsby-plugin-styled-components: ^5.15.0 => 5.15.0
    gatsby-source-cloudinary: ^0.1.14 => 0.1.14
    gatsby-source-filesystem: ^4.15.0 => 4.15.0
    gatsby-transformer-sharp: ^4.15.0 => 4.15.0
  npmGlobalPackages:
    gatsby-cli: 4.15.0
```
